packages/venia-ui/lib/RootComponents/Product/product.js

```diff
@@ -23,6 +23,8 @@ const Product = () => {
 
     const { error, loading, product } = talonProps;
 
+    if (loading && !product) return fullPageLoadingIndicator;
+    if (error && !product) return <ErrorView />;
     if (!product) {
         if (loading) return fullPageLoadingIndicator;
         else if (error) {
```

Comment:
Theses two gates are unreachable now. I think we should have just accepted `develop` in this resolution, eg.

```
if (loading && !product) return fullPageLoadingIndicator;
if (error && !product) return <ErrorView />;
if (!product) { // OOS Message }
```

Reply:
You are right @tjwiebell. @sirugh what do you think should be the result if there is an error?

In `develop` we are rendering

```jsx
<ErrorView />
```

And in `master` we are rendering 

```jsx
<div>
     <FormattedMessage
            id={'product.errorFetch'}
            defaultMessage={'Data Fetch Error'}
       />
</div>
```

Reply:
I think you should use what is in `develop`, as @tjwiebell suggested.